lib/factory.rb

```diff
@@ -0,0 +1,105 @@
+# * Here you must define your `Factory` class.
+# * Each instance of Factory could be stored into variable. The name of this variable is the name of created Class
+# * Arguments of creatable Factory instance are fields/attributes of created class
+# * The ability to add some methods to this class must be provided while creating a Factory
+# * We must have an ability to get/set the value of attribute like [0], ['attribute_name'], [:attribute_name]
+#
+# * Instance of creatable Factory class should correctly respond to main methods of Struct
+# - each
+# - each_pair
+# - dig
+# - size/length
+# - members
+# - select
+# - to_a
+# - values_at
+# - ==, eql?
+
+class Factory
+
+  def self.new(*args, &block)
+
+    class_name = args.shift.capitalize if args[0].is_a?(String)
```

Comment:
Move it to separate method

Reply:
And try to use memoization. (https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/)